src/Uno.UI/UI/Xaml/UIElement.macOS.cs

```diff
@@ -254,17 +272,112 @@ public override void MouseDown(NSEvent evt)
 						OriginalSource = this
 					};
 
-					pointerEventIsHandledInManaged = RaiseEvent(PointerEnteredEvent, args);
+					pointerEventIsHandledInManaged = RaisePointerEvent(PointerPressedEvent, args);
```

Comment:
This PR is overall a great improvement of pointer handling on Mac OS, but I think that we could go a bit further by using the shared code of pointer events that is used on all other platforms. Doing so we will be able to get all gestures (especially the `Tapped` event) and manipulations events.

I'm not able to test it right now, but I think it should works pretty easily with your other changes 😃.

To do that, you should only have to remove all `IsPointerPressed = true|false;`, `IsPointerOver = true|false;` and here instead of directly raising the event, you should invoke [`OnNativePointerEnter(args)`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L399) and equivalent for other `MouseXXX` overrides.

Reply:
I shall try it tomorrow :-)

Reply:
Great 😃 Not sure if you saw that I did a commit that should helps you to understand what I mean https://github.com/dr1rrb/Uno/commit/2c426732893cbb035c9cda3f193313d43b4fd7be , I only forgot to remove at line 138:
```cs
IsPointerPressed = false;
IsPointerOver = false;
```

Reply:
Definitely keeping it in mind 👍👍

Reply:
Done :-)